### PR TITLE
feat: Grid SoundCloud share url parsing

### PR DESCRIPTION
**Pull request by nastita**
### Ticket ID
DEV-11781

### Description
Improve Soundcloud input parsing to accept share urls

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-11781 Support SoundCloud share links](https://app.clickup.com/t/2645698/DEV-11781)

File: domains/grid/utils/gridParser.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-export const parsePlatformInput = (
+export const parsePlatformInput = async (
   platform: GridWidgetType,
   input: string
-): LayoutItemExtended | never => {
+): Promise<LayoutItemExtended> => {
```
Comment (dzbo) [inline domains/grid/utils/gridParser.ts:4]: I think you miss await in the function. Also why `never` type was removed?

File: domains/grid/utils/gridParser.ts [hunk 2]
```diff
@@ -108,21 +108,44 @@ const parseSpotifyWidgetInput = (input: string): LayoutItemExtended | never => {
...
+
+    if (soundCloudEmbed) {
+      return parseSoundCloudWidgetInputFromEmbed(soundCloudEmbed)
+    }
+  }
+
+  throw new Error('Invalid SoundCloud input')
+}
+
+export const parseSoundCloudWidgetInputFromEmbed = (
```
Comment (dzbo) [inline domains/grid/utils/gridParser.ts:135]: I guess we don't need to export this as it's just for internal use.

File: domains/grid/utils/gridParser.ts [hunk 3]
```diff
@@ -132,6 +155,24 @@ const parseSoundCloudWidgetInput = (
   throw new Error('Invalid SoundCloud input')
 }
 
+export const getSoundCloudEmbedUrl = async (
```
Comment (dzbo) [inline domains/grid/utils/gridParser.ts:158]: Same as above.

File: domains/grid/utils/gridParser.ts (hunk 2 above)
Comment (nastita) [reply domains/grid/utils/gridParser.ts:135]: its exported because its covered by some tests

File: domains/grid/utils/gridParser.ts (hunk 1 above)
Comment (nastita) [reply domains/grid/utils/gridParser.ts:4]: good catch

**Review by dzbo — APPROVED** [review]

**Pull request merged**